README.md

```diff
@@ -1,2 +1,145 @@
 # tally-cpp
-C++ tally client
+
+Fast, buffered, hierarchical stats collection in C++.
+
+## Abstract
+
+Tally provides a common interface for emitting metrics, while letting you not worry about the
+velocity of metrics emission.
+
+By default it buffers counters, gauges and histograms at a specified interval, but does not
+buffer timer values.  This is primarily so timer values can have all their values sampled
+if desired and if not they can be sampled as summaries or histograms independently by a reporter.
+
+## Structure
+
+- **Scope**: Keeps track of metrics and their common metadata.
+- **Metrics**: Counters, Gauges, Timers and Histograms.
+- **Reporter**: Implemented by you. Forwards aggregated values from a scope to your metrics
+ingestion pipeline. A default implementation for emitting tagged metrics using Thrift is provided.
+
+### Usage
+
+```c++
+#include <chrono>
+#include <map>
+#include <memory>
+#include <string>
+#include <thread>
+
+#include <m3/reporter.h>
+#include <m3/reporter_builder.h>
+#include <tally/buckets.h>
+#include <tally/scope.h>
+#include <tally/scope_builder.h>
+
+int main(int argc, char** argv) {
+  // Create a reporter that will be used to emit your metrics.
+  auto reporter =
```

Comment:
maybe make it clear that by default people can do 

```
 auto reporter = tally::m3::ReporterBuilder()
          .common_tags({{"service", "statsdex"}, {"env", "production"}})
          .Build();
```

concern is that people will copy/paste from the README, so it should be the simplest version possible (or list multiple versions if you feel strongly)

Reply:
also, could you change `common_tags(...)` to `CommonTags(...)`

Reply:
Err, I think it would be preferable to leave the port and host in. I think ATG will have a wrapper anyway, so I'd prefer to be explicit about the endpoint in the README.

In regards to `common_tags` -> `CommonTags` I was just following the convention of getters and setters being lowercase so I don't think we want to change that.